src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,54 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.commands.Command;
+import duke.parser.Parser;
+import duke.parser.UnableToParseException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import javafx.stage.Stage;
+
+/**
+ * This class is the main driver of the Duke app.
+ */
+public class Duke {
+    private Storage storage;
+    private Ui ui;
+    private TaskList tasks;
+
+    Duke(Stage stage) {
+        try {
+            this.ui = new Ui(stage, this);
+            ui.print("Initializing Duke...");
+            this.storage = new Storage();
+            this.tasks = this.storage.load();
```

Comment:
Very very minor, but I think the use of the ```this``` keyword can be avoided here, as per [this](https://se-education.org/guides/conventions/java/index.html#variables).